### ISSUE TITLE
`this.cacheHint` is set after type check

### ISSUE THREAD

**Issue by poppa** [keystonejs/keystone#1864]
It seems that `this.cacheHint` in the List class is set after it has been type checked:

```
235    if (!['object', 'function', 'undefined'].includes(typeof this.cacheHint)) {
236      throw new Error(`List ${label}'s cacheHint must be an object or function`);
237    }
238    this.cacheHint = cacheHint;
```

In this case it looks to me that `this.cacheHint` will always be `undefined` in the check. I guess the intention on line `235` is to check `cacheHint` and not `this.cacheHint`?

**Comment by Vultraz** [comment]
Looks that way to me too.

**Comment by timleslie** [comment]
Good catch, thanks for the bug report, I'll fix this in #1873 👍
